uber/site_sections/summary.py

```diff
@@ -175,12 +175,101 @@ def food_eligible(self, session):
+                    elif isinstance(col.type, MultiChoice):
+                        # the export has labels separated by ' / ' and we want to convert that
+                        # back into a comma-separate list of integers
+                        label_lookup = {val: key for key, val in col.type.choices}
+                        vals = [label_lookup[label] for label in val.split(' / ')]
+                        val = ','.join(map(str, vals))
+                    elif isinstance(col.type, UTCDateTime):
+                        # we'll need to make sure we use whatever format string we used to
+                        # export this date in the first place
+                        try:
+                            val = UTC.localize(datetime.strptime(val, date_format + ' %H:%M:%S'))
+                        except:
+                            val = UTC.localize(datetime.strptime(val, date_format))
+                    elif isinstance(col.type, Date):
+                        val = datetime.strptime(val, date_format).date()
+                    elif isinstance(col.type, Integer):
+                        val = int(val)
+
+                    # now that we've converted val to whatever it actually needs to be, we
+                    # can just set it on the attendee
+                    setattr(model_instance, colname, val)
+
+                try:
+                    session.commit()
+                except:
+                    log.error('ImportError', exc_info=True)
+                    session.rollback()
+                    message = 'Import unsuccessful'
+
+                id_list.append(model_instance.id)
+
+            if id_list and model == Attendee:
+                all_instances = session.query(model).filter(model.id.in_(id_list)).all()
+
+        return {
+            'message': message,
+            'tables': sorted([model.__name__.lower() for model in Session.all_models()]),
+            'attendees': all_instances
+        }
+
+    def valid_attendees(self):
+        raise HTTPRedirect('export_model?selected_model=attendee')
+
+    def csv_export(self, message='', **params):
+        if 'model' in params:
+            self.export_model(selected_model=params['model'])
+
+        return {
+            'message': message,
+            'tables': sorted([model.__name__.lower() for model in Session.all_models()])
+        }
+
     @csv_file
-    def valid_attendees(self, out, session):
-        cols = [getattr(Attendee, col.name) for col in Attendee.__table__.columns]
+    def export_model(self, out, session, selected_model='', *filters):
+        model = session.get_class_by_tablename(selected_model, 'csv_export')
+
+        cols = [getattr(model, col.name) for col in model.__table__.columns]
         out.writerow([col.name for col in cols])
 
-        for attendee in session.valid_attendees().filter(Attendee.first_name != '').order_by(Attendee.badge_num).all():
+        for attendee in session.query(model).filter(*filters).all():
```

Comment:
Although we have space here for *filters, they aren't actually passed through yet.